docs/howto/profiling.md

```diff
@@ -0,0 +1,156 @@
+# Profiling
+
+Software profiling allows you to inspect the performance of a piece of software, seeing where the bottlenecks are and how they can be improved.
+gperftools is a software profiling package. It contains a CPU profiler, thread-caching malloc library, memory leak detection tool, memory allocation profiler and pprof (discussed later). More information about gperftools and its components can be found on the project's [GitHub](https://github.com/gperftools/gperftools) and [documentation page](https://gperftools.github.io/gperftools/).
+
+## Install gperftools
+
+### Ubuntu
+
+If you're using Ubuntu you can use the following command to install gperftools:
+
+```
+apt install google-perftools libgoogle-perftools-dev
+```
+
+### Other Systems
+
+Alternatively, you can use the following commands to install it:
+
+```
+git clone https://github.com/gperftools/gperftools
+cd gperftools
+git tag -l # checkout the latest release version
+git checkout <gperftools-X.x>
+./autogen.sh
+./configure
+make prefix=/usr libdir=<your/desired/install/dir>
+make prefix=/usr libdir=<your/desired/install/dir> install
+```
+
+If you wish to install gperftools to a directory that is not one of the standard directories for libraries and therefore not findable by the `-l` compiler flag, you will need to specify the path to it with the `GPERF_INSTALL_DIR` option at build time.
+Further information about installing gperftools is [here](https://github.com/gperftools/gperftools/blob/master/INSTALL).
+
+## pprof
+
+pprof is a tool for visualising and analysing profiling data.
+An older version of pprof comes bundled with gperftools but using the newer Go version comes with several benefits: nicer looking graphs and additional options that make looking through specific sections of a program easier being among them.
+
+### Install Go pprof (Optional)
+
+First, you must install Go. Instructions to do so are available [here](https://go.dev/doc/install).
+Optionally, you can install [Graphviz](http://www.graphviz.org/download/) to produce visualisations of profiles.
+
+Then, run the following command to install pprof itself:
+
+```
+go install github.com/google/pprof@latest
+```
+
+## Link gperftools Libraries When Compiling
+
+The library needed to run the CPU profilier should be linked into the ACTS project using the following build option:
+
+```
+-DACTS_ENABLE_CPU_PROFILING=ON
+```
+
+Similarly, to enable the memory profiler the following build option should be used:
+
+```
+-DACTS_ENABLE_MEMORY_PROFILING=ON
+```
+
+Alternatively, you can avoid rebuiding the project by pointing the LD_PRELOAD environment variable to the profiler library for CPU profiling:

```

Comment:
Since this is not the recommended method (but still good to know), can you put this in a sub-sub-section, eg.

### Alternatives to recompiling

so people who read too quickly know they don't have to do both.